app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyTextAreaQuestionScreen.kt

```diff
@@ -9,32 +9,45 @@ import androidx.compose.runtime.Composable
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
 import androidx.compose.runtime.setValue
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.focus.onFocusChanged
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import com.kks.nimblesurveyjetpackcompose.model.SurveyAnswer
 import com.kks.nimblesurveyjetpackcompose.ui.presentation.common.SurveyText
 import com.kks.nimblesurveyjetpackcompose.ui.theme.White20
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+private const val DELAY_VALUE_CHANE_MILLI = 300L
 
 @Composable
 fun SurveyTextAreaQuestionScreen(
     answers: List<SurveyAnswer>,
-    shortText: String,
-    onChooseAnswer: (answers: List<SurveyAnswer>) -> Unit
+    onAnswerChange: (answers: List<SurveyAnswer>) -> Unit
 ) {
-    var answer by remember { mutableStateOf("") }
+    var answer by remember { mutableStateOf(answers.first().answer) }
+    val coroutineScope = rememberCoroutineScope()
+    var debounceJob: Job? = null
```

Comment:
We can make use of `LaunchedEffect` here so that we don't have to manually manage our job 😄 
```suggestion
    LaunchedEffect(key1 = answer) {
        delay()
        onAnswerChange()
    }
```